src/client.js

```diff
@@ -64,10 +64,15 @@ function Client(manager, name, config) {
 		sockets: manager.sockets,
 		manager: manager
 	});
+
 	var client = this;
-	crypto.randomBytes(48, function(err, buf) {
-		client.token = buf.toString("hex");
-	});
+
+	if (!client.config.token) {
+		client.updateToken(function() {
+			client.manager.updateUser(client.name, {token: client.config.token});
+		});
```

Comment:
Just curious, does this really need to be asynchronous? I'm not sure performance would be any affected if sync, but code would look a tiny bit nicer IMO. Nothing ground-breaking of course.